tests/Tests.AzureAppConfiguration/Integration/IntegrationTests.cs

```diff
@@ -0,0 +1,1923 @@
+            // Create a snapshot with the test keys
+            var settingsToInclude = new List<ConfigurationSettingsFilter>
+            {
+                new ConfigurationSettingsFilter($"{testContext.KeyPrefix}:*")
+            };
+
+            ConfigurationSnapshot snapshot = new ConfigurationSnapshot(settingsToInclude);
+
+            snapshot.SnapshotComposition = SnapshotComposition.Key;
+
+            try
+            {
+                // Create the snapshot
+                CreateSnapshotOperation operation = await _configClient.CreateSnapshotAsync(WaitUntil.Completed, snapshotName, snapshot);
+
+                // Track created snapshot for cleanup
+                _createdSnapshotNames.Add(snapshotName);
+
+                return operation.Value.Name;
+            }
+            catch (RequestFailedException ex)
+            {
+                Console.WriteLine($"Error creating snapshot: {ex.Message}");
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Initialize clients to connect to the existing App Configuration store and Key Vault.
+        /// </summary>
+        public async Task InitializeAsync()
+        {
+            try
+            {
+                var credential = GetCredential();
+                string subscriptionId = GetCurrentSubscriptionId();
+
+                // Initialize Azure Resource Manager client
+                var armClient = new ArmClient(credential);
+                SubscriptionResource subscription = armClient.GetSubscriptionResource(new ResourceIdentifier($"/subscriptions/{subscriptionId}"));
+
+                ResourceGroupResource resourceGroup = await subscription.GetResourceGroups().GetAsync(ResourceGroupName);
+
+                AppConfigurationStoreResource appConfigStore = null;
+                KeyVaultResource keyVault = null;
+
+                try
+                {
+                    // Get App Configuration store directly using the resource group and store name
+                    appConfigStore = await resourceGroup.GetAppConfigurationStores().GetAsync(AppConfigStoreName);
+                }
+                catch (RequestFailedException ex) when (ex.Status == 404)
+                {
+                    throw new InvalidOperationException($"App Configuration store '{AppConfigStoreName}' not found in resource group '{ResourceGroupName}'. Please create it before running tests.", ex);
+                }
+
+                _appConfigEndpoint = new Uri(appConfigStore.Data.Endpoint);
+
+                // Get connection string from the store
+                var accessKeys = appConfigStore.GetKeysAsync();
+                var primaryKey = await accessKeys.FirstOrDefaultAsync();
+
+                if (primaryKey == null)
+                {
+                    throw new InvalidOperationException("Failed to retrieve access keys from App Configuration store.");
+                }
+
+                _connectionString = primaryKey.ConnectionString;
```

Comment:
```suggestion
                var primaryKey = await accessKeys.FirstOrDefaultAsync();

                if (primaryKey == null)
                {
                    throw new InvalidOperationException("Failed to retrieve access keys from App Configuration store.");
                }

                _connectionString = (await accessKeys.FirstAsync()).ConnectionString;
```

Reply:
Not sure I understand, `FirstAsync` instead of `FirstOrDefaultAsync`?

Reply:
Yes, FirstAsync has the invalid operation exception built-in.

Reply:
Oh I see thanks